zaza/openstack/charm_tests/mysql/tests.py

```diff
@@ -670,3 +705,115 @@ def test_999_migrate_percona_to_mysql(self):
         test_config = lifecycle_utils.get_charm_config(fatal=False)
         zaza.model.wait_for_application_states(
             states=test_config.get("target_deploy_status", {}))
+
+
+class MySQLInnoDBClusterScaleTest(MySQLBaseTest):
+    """Percona Cluster cold start tests."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running mysql-innodb-cluster scale tests."""
+        super().setUpClass()
+        cls.application = "mysql-innodb-cluster"
+        cls.test_config = lifecycle_utils.get_charm_config(fatal=False)
+        cls.states = cls.test_config.get("target_deploy_status", {})
+
+    def test_800_remove_leader(self):
+        """Remove leader node.
+
+        We start with a three node cluster, remove one, down to two.
+        The cluster will be in waiting state.
+        """
+        logging.info("Scale in test: remove leader")
+        leader, nons = self.get_leaders_and_non_leaders()
+        leader_unit = zaza.model.get_unit_from_name(leader)
+        zaza.model.destroy_unit(self.application_name, leader)
+
+        # XXX time.sleep roundup
+        # https://github.com/openstack-charmers/zaza-openstack-tests/issues/46
+        # Without this the idle time check returns too fast, and when the next
+        # test runs we end up with a new node becoming leader.
+        time.sleep(60)
+
+        logging.info("Wait till model is idle ...")
+        zaza.model.block_until_all_units_idle()
+
+        logging.info(
+            "Removing old unit from cluster: {} "
+            .format(leader_unit.public_address))
+        action = zaza.model.run_action(
+            nons[0],
+            "remove-instance",
+            action_params={
+                "address": leader_unit.public_address,
+                "force": True})
+        assert action.data.get("results") is not None, (
+            "Remove instance action failed: No results: {}"
+            .format(action.data))
```

Comment:
Should we wait at this point for the model to stabilise before moving on to the next test?

Reply:
This action only affects the metadata. We already waited for the model change above.